src/main/java/command/UserCommand.java

```diff
@@ -0,0 +1,57 @@
+package command;
+
+import exception.InvalidCommandException;
+import task.Storage;
+import task.TaskList;
+import ui.Ui;
+
+/**
+ * The UserCommand interface deals with executing user commands
+ */
+public abstract class UserCommand {
+
+    /**
+     * Creates the appropriate UserCommand based on the command name
+     *
+     * @param commandName String representing the name of the user command
+     * @return UserCommand to be executed
+     */
+    public static UserCommand toCommand(String commandName) throws InvalidCommandException {
+        switch (commandName) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+        case "unmark":
+            return new UpdateStatusCommand();
+        case "todo":
+        case "deadline":
+        case "event":
+            return new AddCommand();
+        case "delete":
+            return new DeleteCommand();
+        case "find":
+            return new FindCommand();
+        default:
+            throw new InvalidCommandException();
+        }
+    }
+
+    /**
+     * Checks whether to continue running the program
+     * @return Boolean of whether the command is an exit command
+     */
+    public boolean continueRunning() {
```

Comment:
It will be better if the method name can sound like a boolean, eg isContinueRunning

(ref: [see here](https://se-education.org/guides/conventions/java/index.html#:~:text=Boolean%20variables/methods%20should%20be%20named%20to%20sound%20like%20booleans))